lib/sass/rails/railtie.rb

```diff
@@ -55,8 +55,13 @@ class Railtie < ::Rails::Railtie
 
       config.assets.configure do |env|
         if env.respond_to?(:register_engine)
-          env.register_engine '.sass', Sass::Rails::SassTemplate, silence_deprecation: true
-          env.register_engine '.scss', Sass::Rails::ScssTemplate, silence_deprecation: true
+          args = ['.sass', Sass::Rails::SassTemplate]
+          args << { silence_deprecation: true } if env.method(:register_engine).arity > 2
+          env.register_engine *args
```

Comment:
I guess we'd better put brackets here or we'll get `warning: *' interpreted as argument prefix`